src/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -1631,127 +1631,34 @@ private boolean hasSpecialFormMapping(String textID, String locale) {
         return false;
     }
 
-    protected DataBinding processStandardBindAttributes( List<String> usedAtts, Element e) {
-        usedAtts.add(ID_ATTR);
-        usedAtts.add(NODESET_ATTR);
-        usedAtts.add("type");
-        usedAtts.add("relevant");
-        usedAtts.add("required");
-        usedAtts.add("readonly");
-        usedAtts.add("constraint");
-        usedAtts.add("constraintMsg");
-        usedAtts.add("calculate");
-        usedAtts.add("preload");
-        usedAtts.add("preloadParams");
-
-        DataBinding binding  = new DataBinding();
-
-
-        binding.setId(e.getAttributeValue("", ID_ATTR));
-
-        String nodeset = e.getAttributeValue(null, NODESET_ATTR);
-        if (nodeset == null) {
-            throw new XFormParseException("XForm Parse: <bind> without nodeset",e);
-        }
-        IDataReference ref;
-        try {
-            ref = new XPathReference(nodeset);
-        } catch(XPathException xpe) {
-            throw new XFormParseException(xpe.getMessage());
-        }
-        ref = getAbsRef(ref, _f);
-        binding.setReference(ref);
-
-        binding.setDataType(getDataType(e.getAttributeValue(null, "type")));
-
-        String xpathRel = e.getAttributeValue(null, "relevant");
-        if (xpathRel != null) {
-            if ("true()".equals(xpathRel)) {
-                binding.relevantAbsolute = true;
-            } else if ("false()".equals(xpathRel)) {
-                binding.relevantAbsolute = false;
-            } else {
-                Condition c = buildCondition(xpathRel, "relevant", ref);
-                c = (Condition)_f.addTriggerable(c);
-                binding.relevancyCondition = c;
-            }
-        }
+    private final List<String> usedAtts = Collections.unmodifiableList(Arrays.asList(
```

Comment:
A fun fact: In some immutable Set implementations, when given a small set of values, no actual hash set or tree set is created. Instead the elements are saved in an array or something similar.